### ISSUE TITLE
haskell-ide-engine broken

### ISSUE THREAD

**Issue by Infinisil** [NixOS/nixpkgs#34447]
## Issue description
The recently added haskell-ide-engine is now broken (already marked as broken by @peti in https://github.com/NixOS/nixpkgs/commit/b83990dd76c812632509a7efa4caf90da277236e). This new failure is briefly discussed in #32947's [comments](https://github.com/NixOS/nixpkgs/pull/32947#issuecomment-359491863). The full build logs (of the failing dependency path) are [here](https://gist.github.com/Infinisil/0964bb60bfb2d7bba2002c71aa3df8ba) and these are the errory bits

```
Warning:
    This package indirectly depends on multiple versions of the same package. This is very likely to cause a compile failure.
      package ghc-mod-core (ghc-mod-core-5.9.0.0) requires haskell-src-exts-1.19.1-GkJUFo8Rp3b1KlAdoTXU6c
      package hlint (hlint-2.0.15-DnfSXCOnlvhJh25E4Cb1Bg) requires haskell-src-exts-1.20.1-835K5nW7Qg0K3DUFrUYhiW
      package haskell-src-exts-util (haskell-src-exts-util-0.2.1.2-4UhAxVuyQajDfxD9gJGXOC) requires haskell-src-exts-1.20.1-835K5nW7Qg0K3DUFrUYhiW
Using Cabal-2.0.1.0 compiled by ghc-8.2
Using compiler: ghc-8.2.2
Using install prefix:
/nix/store/lfi038zbjrl64f2zpk72bwsq2b26yn95-ghc-mod-core-5.9.0.0
Executables installed in:
/nix/store/lfi038zbjrl64f2zpk72bwsq2b26yn95-ghc-mod-core-5.9.0.0/bin
Libraries installed in:
/nix/store/lfi038zbjrl64f2zpk72bwsq2b26yn95-ghc-mod-core-5.9.0.0/lib/ghc-8.2.2/ghc-mod-core-5.9.0.0
Dynamic Libraries installed in:
/nix/store/lfi038zbjrl64f2zpk72bwsq2b26yn95-ghc-mod-core-5.9.0.0/lib/ghc-8.2.2/x86_64-linux-ghc-8.2.2
Private executables installed in:
/nix/store/lfi038zbjrl64f2zpk72bwsq2b26yn95-ghc-mod-core-5.9.0.0/libexec/x86_64-linux-ghc-8.2.2/ghc-mod-core-5.9.0.0
Data files installed in:
/nix/store/lfi038zbjrl64f2zpk72bwsq2b26yn95-ghc-mod-core-5.9.0.0/share/x86_64-linux-ghc-8.2.2/ghc-mod-core-5.9.0.0
Documentation installed in:
/nix/store/cx5cf3l7k452dan6j4cicyw0fvxpj1xb-ghc-mod-core-5.9.0.0-doc/share/doc
Configuration files installed in:
/nix/store/lfi038zbjrl64f2zpk72bwsq2b26yn95-ghc-mod-core-5.9.0.0/etc
No alex found
Using ar found on system at:
/nix/store/mdyy001q67hiks0g24ra53z7ckm4jfr4-binutils-2.28.1/bin/ar
No c2hs found
Using cpphs version 1.20.8 found on system at:
/nix/store/r4kakwazf0n9kwk1vnqs24ky5ppsh4m0-cpphs-1.20.8/bin/cpphs
No doctest found
Using gcc version 6.4.0 given by user at:
/nix/store/6s3dl13ingr21z93dy6zxcc652wdlcrh-gcc-wrapper-6.4.0/bin/gcc
Using ghc version 8.2.2 found on system at:
/nix/store/75hyhwqmh2zh2wflffn1qgrrhy5nl8vg-ghc-8.2.2/bin/ghc
Using ghc-pkg version 8.2.2 found on system at:
/nix/store/75hyhwqmh2zh2wflffn1qgrrhy5nl8vg-ghc-8.2.2/bin/ghc-pkg
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.18.1 found on system at:
/nix/store/75hyhwqmh2zh2wflffn1qgrrhy5nl8vg-ghc-8.2.2/bin/haddock
No happy found
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.67 found on system at:
/nix/store/75hyhwqmh2zh2wflffn1qgrrhy5nl8vg-ghc-8.2.2/bin/hpc
Using hsc2hs version 0.68.2 found on system at:
/nix/store/75hyhwqmh2zh2wflffn1qgrrhy5nl8vg-ghc-8.2.2/bin/hsc2hs
Using hscolour version 1.24 found on system at:
/nix/store/wx43l198z7ckkifwzm69bbaam3djmnv3-hscolour-1.24.2/bin/HsColour
No jhc found
Using ld found on system at:
/nix/store/i8fxs7ppvkayaczq1vwzw9vcqkqrrj02-binutils-wrapper-2.28.1/bin/ld
No lhc found
No lhc-pkg found
No pkg-config found
Using runghc version 8.2.2 found on system at:
/nix/store/75hyhwqmh2zh2wflffn1qgrrhy5nl8vg-ghc-8.2.2/bin/runghc
No shelltest found
Using strip version 2.28 found on system at:
/nix/store/mdyy001q67hiks0g24ra53z7ckm4jfr4-binutils-2.28.1/bin/strip
Using tar found on system at:
/nix/store/zr5z057g5ilp885mp2mlnq2x9djp386w-gnutar-1.30/bin/tar
No uhc found
*** abort because of serious configure-time warning from Cabal
builder for '/nix/store/5in0333j40sfl82x6rwhj13hadh7wj9w-ghc-mod-core-5.9.0.0.drv' failed with exit code 1
cannot build derivation '/nix/store/bg9ay2zyyk1a03smz9lf9xmhfld4x5kf-HaRe-0.8.4.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/4526zp7j262fd7pmh15bnppa5dgim2i7-ghc-mod-5.9.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/3jsrjdw9jqagxi3fa441sgzlr6dpkxq2-hie-plugin-api-0.1.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/bsm5nmr3kypmi6f3hp998jp0m8xqmgs1-hie-apply-refact-0.1.0.0.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/1w2yw4q7ay35cmrpdwxhqyvvddybd14q-hie-brittany-0.1.0.0.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/31v2dbq37c44060lvdascky8vmw07anp-hie-build-plugin-0.1.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/1g1j3kvarc05vnlixfj0vq2yfh8fpkhn-hie-eg-plugin-async-0.1.0.0.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/8vdj7ahhr9mrzyf49r8jchvvpbxrvr2q-hie-example-plugin2-0.1.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/x25iqkr5i4ad16n9x37616kph33pk4f0-hie-ghc-mod-0.1.0.0.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/pbrxwagc9ic6zpi2b7zk5bpb61lkac8a-hie-ghc-tree-0.1.0.0.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/w6jhhzl958k2riqpkavrs39d40n8j0z0-hie-hoogle-0.1.0.0.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/y1828lrsjfpgjnqrlk58g61knc3n63c0-hie-hare-0.1.0.0.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/9pyn7r9lbk06lchkxfm3xpfq6zprpy1f-hie-haddock-0.1.0.0.drv': 6 dependencies couldn't be built
cannot build derivation '/nix/store/j77zb9fpq8vqpiqx87az65bsh672lrpk-haskell-ide-engine-0.1.0.0.drv': 12 dependencies couldn't be built
error: build of '/nix/store/j77zb9fpq8vqpiqx87az65bsh672lrpk-haskell-ide-engine-0.1.0.0.drv' failed
```

@shlevy @michalrus @Fresheyeball

### Steps to reproduce
    
    nix-build -A haskellPackages.haskell-ide-engine --arg config '{ allowBroken = true; }'

## Technical details

nixpkgs version: fefa9ef756298e6863ece2e93bd057124b78452f

**Comment by Infinisil** [comment]
Pinging @domenkozar as well

**Comment by domenkozar** [comment]
I'm very much not in favor or how this is implemented. I think it should live outside of nixpkgs, I plan to provide such a repository soon with support for https://github.com/input-output-hk/stack2nix

cc @shlevy for current package

**Comment by shlevy** [comment]
@domenkozar What's wrong with having this in nixpkgs? I'm already trying to work with @peti on a way to automate things like this so it doesn't need to be manually maintained, but in the mean time we manually maintain thousands of packages in nixpkgs...

**Comment by domenkozar** [comment]
All I'm saying is that it doesn't have to be maintained manually and I think it shouldn't - see amount of recent breakages.

**Comment by domenkozar** [comment]
Here we go: https://github.com/domenkozar/hie-nix

**Comment by peti** [comment]
I have to say, the fact that almost all those HIE packages were broken within 2 weeks after their addition and that nobody seems to bother fixing them does suggest that this code should probably not be in Nixpkgs.

**Comment by shlevy** [comment]
@peti I haven't fixed them because I was trying to approach this through https://github.com/NixOS/cabal2nix/issues/317 ...

**Comment by peti** [comment]
I don't see how that might be connected. The build issue reported here is not a matter of "updating" or "re-generating" the package set, which is what that ticket is about.

**Comment by shlevy** [comment]
But this issue could be fixed by updating to the latest for hie and regenerating, and generating the extra deps needed automatically as well if necessary.

**Comment by peti** [comment]
I guess ... still, it doesn't bode well for the current hie package set, which is clearly broken and apparently will remain broken for the foreseeable future. My impression is that it might be better for Nixpkgs users to give https://github.com/domenkozar/hie-nix a try.

**Comment by mpickering** [comment]
Domen's hie-nix works well for me.

**Comment by Infinisil** [comment]
hie-nix works well, @domenkozar! What should be done about the current `haskell-ide-engine`? Maybe we could replace it with something like

```nix
throw ''
  Due to its constant breakage, haskell-ide-engine is currently only well
  supported unofficially through https://github.com/domenkozar/hie-nix
''
```

**Comment by shlevy** [comment]
I'll work on a maintainable fix or remove it before the 18.03 release.

**Comment by michalrus** [comment]
Another vote, hie-nix working very well! Thank you, @domenkozar. (=

**Comment by domenkozar** [comment]
@shlevy 18.03 is due in a week, maybe it's best to focus efforts on pinned hie-nix?

**Comment by shlevy** [comment]
Yep, will remove.

**Comment by shlevy** [comment]
Fixed in ead2ba88097c58f60cdd0ae5475c64192c7c21d3 and 9e804ba5bfa245c2b3ecff96161ca806d91c5d2b
